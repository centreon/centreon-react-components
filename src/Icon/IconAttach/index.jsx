/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable react/prop-types */

import React from 'react';

import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  icon: {
    color: '#707070',
    cursor: 'initial',
    fontSize: 14,
    verticalAlign: 'middle',
  },
  iconAttach: {
    width: 80,
    height: 49,
    textAlign: 'center',
    lineHeight: '46px',
    backgroundColor: '#fff',
    display: 'inline-block',
    marginRight: 15,
    verticalAlign: 'middle',
    cursor: 'initial',
    marginLeft: -20,
  },
  iconAttachLabel: {
    fontSize: 12,
    color: '#707070',
    display: 'inline-block',
    verticalAlign: 'middle',
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    marginLeft: 3,
  },
  iconAttachImage: {
    maxWidth: '100%',
    maxHeight: '100%',
    verticalAlign: 'middle',
  },
}));

const IconAttach = ({
  onClick,
  defaultImage,
  uploadedImage,
  imgSource,
  title,
  labelNoIcon = 'NO ICON',
}) => {
  const classes = useStyles();

  return (
    <span onClick={onClick} className={classes.iconAttach}>
      {defaultImage && (
        <>
          <span className={classes.iconAttachLabel}>{labelNoIcon}</span>
        </>
      )}
      {uploadedImage && (
        <img
          src={imgSource}
          className={classes.iconAttachImage}
          alt={title}
          title={title}
        />
      )}
    </span>
  );
};

export default IconAttach;
