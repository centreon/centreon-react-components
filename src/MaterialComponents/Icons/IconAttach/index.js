/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */

import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Attach from '@material-ui/icons/AttachFile';

const useStyles = makeStyles(() => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  icon: {
    color: '#707070',
    cursor: 'pointer',
    fontSize: 14,
    verticalAlign: 'middle',
  },
  iconAttach: {
    width: 80,
    height: 49,
    textAlign: 'center',
    lineHeight: '49px',
    backgroundColor: '#fff',
    display: 'inline-block',
    marginRight: 15,
    verticalAlign: 'middle',
    cursor: 'pointer',
    marginLeft: -20,
  },
  iconAttachLabel: {
    fontSize: 12,
    color: '#707070',
    display: 'inline-block',
    verticalAlign: 'middle',
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    marginLeft: 3,
  },
}));

function IconAttach({ customStyle, onClick }) {
  const classes = useStyles();

  return (
    <React.Fragment>
      <span onClick={onClick} className={classes.iconAttach}>
        <Attach style={customStyle} className={classes.icon} />
        <span className={classes.iconAttachLabel}>ICON</span>
      </span>
    </React.Fragment>
  );
}

export default IconAttach;
