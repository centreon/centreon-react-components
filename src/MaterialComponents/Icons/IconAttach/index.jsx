import React from 'react';
import Attach from '@material-ui/icons/AttachFile';
import styled from '@emotion/styled';
import MaterialIcon from '../MaterialIcon';

const AttachMaterial = styled(MaterialIcon)(() => ({
  width: 80,
  height: 49,
  textAlign: 'center',
  lineHeight: '49px',
  marginRight: 15,
  marginLeft: -20,
}));

const MiniAttach = styled(Attach)(() => ({
  fontSize: 14,
  verticalAlign: 'middle',
}));

const Label = styled.span(() => ({
  fontSize: 12,
  fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
  marginLeft: 3,
}));

function IconAttach(props) {
  return (
    <AttachMaterial {...props}>
      <MiniAttach />
      <Label>ICON</Label>
    </AttachMaterial>
  );
}

export default IconAttach;
