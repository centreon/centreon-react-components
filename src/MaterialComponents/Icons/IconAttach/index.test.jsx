/* eslint-disable no-undef */

import React from 'react';
import { create } from 'react-test-renderer';
import IconAttach from '.';

describe('IconAttach', () => {
  it('renders', () => {
    const wrapper = create(<IconAttach />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
