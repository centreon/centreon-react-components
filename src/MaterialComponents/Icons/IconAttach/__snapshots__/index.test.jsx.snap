// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconAttach renders 1`] = `
.emotion-2 {
  display: inline-block;
  vertical-align: middle;
  height: 24px;
  cursor: pointer;
  color: #707070;
  width: 80px;
  height: 49px;
  text-align: center;
  line-height: 49px;
  margin-right: 15px;
  margin-left: -20px;
}

.emotion-0 {
  font-size: 14px;
  vertical-align: middle;
}

.emotion-1 {
  font-size: 12px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  margin-left: 3px;
}

<span
  className="emotion-2"
>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root emotion-0"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
  <span
    className="emotion-1"
  >
    ICON
  </span>
</span>
`;
